### PR TITLE
Add support for text blob bindings

### LINKED ISSUE #483: Allow file bindings as text_blob
#### Overview and problem statement

We have a worker that does a HTML transformation that adds a new JavaScript script tag, called client code. We want to store the client code along the server-side transformation.

The server-side and client-side codes need to keep sync, even during deployements, so I think KV wouldn't be a good fit.

Alternatively, we can use webpack to inline the client-side code into the worker. But allowing files as text_blob seems useful (a customer requested to inline images, we can use the same technique). 

#### Basic example

worker.js:
```js
addEventListener('fetch', event => {
  event.respondWith(handleRequest(event.request))
})

async function handleRequest(request) {
  return new Response(CLIENT_SIDE_CODE);
}
```

metadata.json:
```json
{
  "body_part": "foo",
  "bindings": [
    {
      "name": "CLIENT_SIDE_CODE",
      "type": "text_blob",
      "part": "CLIENT_SIDE_CODE"
    },
  ]
}
```

Then the multipart from needs to include the `CLIENT_SIDE_CODE` part. Note that the size limit needs to take metadata.json into account too.

### PR DESCRIPTION
**Pull request by xortive**
closes #483 

I need to write docs, and confirm I got the env inheritance stuff right (I copied how it works for vars, need to ensure that is correct.)

But my baby test passes :)

I probably need to ensure size limits are still taken into account here as well.

changelog/docs:

Wrangler now supports text blobs! text blobs are values to use in your workers, but read from a file instead of a string in your TOML.

Usage:

text_blobs = { FOO = "path/to/foo.txt", BAR = "path/to/bar.txt" }

### CODE REVIEW

**Comment by ashleymichal** [comment]
> I probably need to ensure size limits are still taken into account here as well.

correct. there's another ticket for that: #896

File: src/settings/toml/manifest.rs [hunk 1]
```diff
@@ -43,6 +43,7 @@ pub struct Manifest {
     pub kv_namespaces: Option<Vec<ConfigKvNamespace>>,
     pub env: Option<HashMap<String, Environment>>,
     pub vars: Option<HashMap<String, String>>,
+    pub blobs: Option<HashMap<String, String>>,
```
Comment (ashleymichal) [inline src/settings/toml/manifest.rs:46]: you should be able to use a `<HashMap<String, PathBuf>>` explicitly

File: src/settings/toml/manifest.rs [hunk 2]
```diff
@@ -239,6 +241,9 @@ impl Manifest {
 
             // don't inherit vars
             target.vars = environment.vars.clone();
+
+            // don't inherit blobs
+            target.blobs = environment.blobs.clone();
```
Comment (ashleymichal) [inline src/settings/toml/manifest.rs:246]: i don't remember why we chose not to inherit vars... 🤔

**Review by ashleymichal — COMMENTED** [review]
i'm not sure about the name `blobs`. this likely deserves some product attention.

File: src/settings/toml/manifest.rs (hunk 1 above)
Comment (xortive) [reply src/settings/toml/manifest.rs:46]: I'd have used PathBuf, but webpack_config above uses String so I used that, but I can change it if you'd like.

**Comment by ispivey** [comment]
@rita3ko : "Let's call 'em text blobs specifically, and just support those"

File: src/settings/toml/manifest.rs (hunk 1 above)
Comment (ashleymichal) [reply src/settings/toml/manifest.rs:46]: yes please 😄

File: src/settings/toml/manifest.rs (hunk 2 above)
Comment (ashleymichal) [reply src/settings/toml/manifest.rs:246]: i think these should likely be inherited; cc @rita3ko any thoughts?

**Review by ashleymichal — APPROVED** [review]

**Comment by ashleymichal** [comment]
lgtm once the name change is complete.

File: src/settings/toml/manifest.rs (hunk 2 above)
Comment (ashleymichal) [reply src/settings/toml/manifest.rs:246]: these should be inherited i think.

Comment (xortive) [reply src/settings/toml/manifest.rs:246]: My only reason for not inheriting these, is it feels weird to do something different than for vars.

**Comment by ashleymichal** [comment]
@xortive are you in a position to address the above comments or should someone pick this up and carry it over the finish line?

**Comment by xortive** [comment]
> @xortive are you in a position to address the above comments or should someone pick this up and carry it over the finish line?

Sorry I've been unresponsive with this PR, I've got some free time today so I'm going to get this out the door

File: src/settings/toml/manifest.rs (hunk 2 above)
Comment (xortive) [reply src/settings/toml/manifest.rs:246]: @ashleymichal @rita3ko Do y'all have a final opinion here? Y'all have more context than I do about what should/should not get inherited

**Comment by xortive** [comment]
name change done, going to resolve conflicts now

**Comment by xortive** [comment]
Alright, looks like this should be ready to go, unless we choose to inherit text_blobs.

**Review by xtuc — APPROVED** [review]

**Review by nataliescottdavidson — APPROVED** [review]
Could you write some docs / small lil paragraph explaining usage for the changelog?

**Pull request merged**